frontend/pages/reports.js

```diff
@@ -0,0 +1,262 @@
+import s from 'styled-components'
+import { useEffect, useState } from 'react'
+import renderPage from '../renderPage.js'
+import { Icon, Flex, Empty } from '../components/common'
+import { doApiRequest, API_BASE_URL } from '../utils'
+import { Sidebar } from '../components/common/Sidebar'
+import Checkbox, { CheckboxLabel } from '../components/common/Checkbox'
+import { Container, WideContainer } from '../components/common/Container'
+import {
+  CLUBS_GREY,
+  RED,
+  ALLBIRDS_GRAY,
+  MEDIUM_GRAY,
+} from '../constants/colors'
+
+const TallTextArea = s.textarea`
+  height: 6em;
+`
+
+const GroupLabel = s.h4`
+  margin-bottom: 0em !important;
+  font-size: 32px;
+  color: #626572;
+`
+
+const HoverListElement = s.li`
+
+  span {
+    display: none;
+  }
+
+  &:hover span {
+    display: inline;
+    cursor: pointer;
+  }
+
+`
+
+const SelectedManager = ({ value, onClick }) => (
+  <HoverListElement>
+    {value}
+
+    <span style={{ marginLeft: '1em' }} onClick={onClick}>
+      X
+    </span>
+  </HoverListElement>
+)
+
+const Reports = ({ nameToCode }) => {
+  const fields = {
+    Fields: Object.keys(nameToCode),
+  }
+
+  const [nameInput, setNameInput] = useState('')
+  const [descInput, setDescInput] = useState('')
+
+  const [reports, setReports] = useState([])
+  const [reportFlag, updateReportFlag] = useState(false)
+
+  useEffect(() => {
+    doApiRequest('/reports/?format=json')
+      .then(resp => {
+        if (resp.ok) {
+          return resp.json()
+        } else {
+          return []
+        }
+      })
+      .then(data => setReports(data))
+  }, [reportFlag])
+
+  const [includedFields, setIncludedFields] = useState(
+    (() => {
+      const initial = {}
+      Object.keys(fields).forEach(group =>
+        fields[group].forEach(f => {
+          initial[f] = false
+        })
+      )
+      return initial
+    })()
+  )
+
+  const query = {
+    format: 'xlsx',
+    fields: Object.keys(includedFields)
+      .filter(field => includedFields[field])
+      .map(name => nameToCode[name])
+      .filter(e => e !== undefined),
+  }
+
+  const generateCheckboxGroup = (groupName, fields) => {
+    return (
+      <div key={groupName} style={{ flexBasis: '50%', flexShrink: 0 }}>
+        <GroupLabel
+          key={groupName}
+          className="subtitle is-4"
+          style={{ color: CLUBS_GREY }}
+        >
+          {groupName}
+        </GroupLabel>
+        {fields.map((field, idx) => (
+          <div key={idx}>
+            <Checkbox
+              id={field}
+              checked={includedFields[field]}
+              onChange={() => {
+                setIncludedFields(prev => ({ ...prev, [field]: !prev[field] }))
+              }}
+            />
+            {'  '}
+            <CheckboxLabel htmlFor={field}>{field}</CheckboxLabel>
+          </div>
+        ))}
+      </div>
+    )
+  }
+
+  return (
+    <Container>
+      <WideContainer>
```

Comment:
`WideContainer` already has a `Container` inside of it. If it alone is too wide, use `MediumContainer` (I think is what it's called) or just use `Row` and `Col` directly